### PR TITLE
switch to the new coin api on landing page

### PR DESCRIPTION
**Pull request by Jibz1**
- This PR starts the migration away from `getCoins`  to `getAllBalances` and `getBalance`
- Create hooks for each API with refetchInterval of 2 secs may tweak this 
- update the landing page and staking section to use this API 
- remove sass styles
- disable staking for wallet without SUI token

Before
Screenshot 2023-02-22 at 10 24 47 AM

after
Screenshot 2023-02-22 at 5 24 07 PM

### CODE REVIEW

File: apps/wallet/src/ui/app/hooks/useGetAllBalance.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    address: SuiAddress | null;
+};
+export function useGetAllBalance({
+    address,
+}: GetCoinBalanceProps): UseQueryResult<CoinBalance[], unknown> {
+    const rpc = useRpc();
+    const response = useQuery(
+        ['get-all-balance', address],
+        () => rpc.getAllBalances(address!),
+        // refetchInterval is set to 2 seconds
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/hooks/useGetAllBalance.ts:18]: Looks like it's actually 20 seconds in code? I think 2 seconds is a bit too often potentially.

File: apps/wallet/src/ui/app/hooks/useGetAllBalance.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { type CoinBalance, type SuiAddress } from '@mysten/sui.js';
+import { useQuery, type UseQueryResult } from '@tanstack/react-query';
+
+import { useRpc } from '_hooks';
+type GetCoinBalanceProps = {
+    address: SuiAddress | null;
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/hooks/useGetAllBalance.ts:9]: Probably mark this as optional if it's nullable too

File: apps/wallet/src/ui/app/hooks/useGetAllBalance.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+};
+export function useGetAllBalance({
+    address,
+}: GetCoinBalanceProps): UseQueryResult<CoinBalance[], unknown> {
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/hooks/useGetAllBalance.ts:13]: Don't bother typing the return, just let it be inferred out

File: apps/wallet/src/ui/app/hooks/useGetAllBalance.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    const rpc = useRpc();
+    const response = useQuery(
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/hooks/useGetAllBalance.ts:15]: Just directly return here instead of using an intermediate variable.

File: apps/wallet/src/ui/app/hooks/useGetCoinBalance.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+type GetCoinBalanceProps = {
+    address: SuiAddress | null;
+    coinType: string;
+};
+export function useGetCoinBalance({
+    address,
+    coinType,
+}: GetCoinBalanceProps): UseQueryResult<CoinBalance, unknown> {
+    const rpc = useRpc();
+    const response = useQuery(
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/hooks/useGetCoinBalance.ts:17]: Directly return

File: apps/wallet/src/ui/app/hooks/useGetCoinBalance.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+import { useRpc } from '_hooks';
+type GetCoinBalanceProps = {
+    address: SuiAddress | null;
+    coinType: string;
+};
+export function useGetCoinBalance({
+    address,
+    coinType,
+}: GetCoinBalanceProps): UseQueryResult<CoinBalance, unknown> {
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/hooks/useGetCoinBalance.ts:15]: Don't bother with return type

File: apps/wallet/src/ui/app/hooks/useGetCoinBalance.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { type CoinBalance, type SuiAddress } from '@mysten/sui.js';
+import { useQuery, type UseQueryResult } from '@tanstack/react-query';
+
+import { useRpc } from '_hooks';
+type GetCoinBalanceProps = {
+    address: SuiAddress | null;
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/hooks/useGetCoinBalance.ts:9]: Why not have these hooks grab the address itself? Not super opinionated in this, just curious.

File: apps/wallet/src/ui/app/pages/home/tokens/TokensDetails.tsx [hunk 8]
```diff
@@ -31,29 +34,23 @@ const emptyWalletDescription = (
 );
 
 type TokensProps = {
-    allCoinTypes: string[];
-    coinBalance: bigint;
-    balances: Record<string, bigint>;
+    coinBalance: bigint | number;
+    balances: CoinBalanceType[] | [];
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/pages/home/tokens/TokensDetails.tsx:38]: Seems weird to have these passed down like this, why not just have this component fetch?

File: apps/wallet/src/ui/app/pages/home/tokens/TokensDetails.tsx [hunk 9]
```diff
@@ -76,11 +73,25 @@ function MyTokens({
...
+    const activeCoinType = coinType || SUI_TYPE_ARG;
     const accountAddress = useAppSelector(({ account }) => account.address);
-    const balances = useAppSelector(accountAggregateBalancesSelector);
-    const tokenBalance = balances[activeCoinType] || BigInt(0);
-    const allCoinTypes = useMemo(() => Object.keys(balances), [balances]);
+    const { data: coinBalance, isLoading: loadingBalances } = useGetAllBalance({
+        address: accountAddress,
+    });
+
+    const tokenBalance = useMemo(() => {
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/pages/home/tokens/TokensDetails.tsx:82]: Why not have this use the individual fetching route?

File: apps/wallet/src/ui/app/hooks/useGetCoinBalance.ts (hunk 7 above)
Comment (Jibz1) [reply apps/wallet/src/ui/app/hooks/useGetCoinBalance.ts:9]: I think there may be instances once multi-account is active that we want the balance of the address other than the current/active one

File: apps/wallet/src/ui/app/hooks/useGetAllBalance.ts (hunk 1 above)
Comment (Jibz1) [reply apps/wallet/src/ui/app/hooks/useGetAllBalance.ts:18]: I will change it

File: apps/wallet/src/ui/app/pages/home/tokens/TokensDetails.tsx (hunk 9 above)
Comment (Jibz1) [reply apps/wallet/src/ui/app/pages/home/tokens/TokensDetails.tsx:75]: Done

File: apps/wallet/src/ui/app/pages/home/tokens/TokensDetails.tsx (hunk 8 above)
Comment (Jibz1) [reply apps/wallet/src/ui/app/pages/home/tokens/TokensDetails.tsx]: done

File: apps/wallet/src/ui/app/hooks/useGetAllBalance.ts [hunk 10]
```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { type SuiAddress } from '@mysten/sui.js';
+import { useQuery } from '@tanstack/react-query';
+
+import { useRpc } from '_hooks';
+
+export function useGetAllBalance(address?: SuiAddress | null) {
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/hooks/useGetAllBalance.ts:9]: Probably rename this to `useGetAllBalances` to mirror the RPC method.

File: apps/wallet/src/ui/app/hooks/useGetCoinBalance.ts [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+
+import { useRpc } from '_hooks';
+type GetCoinBalanceProps = {
+    address?: SuiAddress | null;
+    coinType: string;
+};
+export function useGetCoinBalance({ address, coinType }: GetCoinBalanceProps) {
+    const rpc = useRpc();
+    return useQuery(
+        // combine address and coinType to make a unique key account for multiple addresses and coins
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/hooks/useGetCoinBalance.ts:15]: Just comma separate them, that's what the point of the array is. You can remove this comment too.

File: apps/wallet/src/ui/app/hooks/useGetCoinBalance.ts [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+
+import { type SuiAddress } from '@mysten/sui.js';
+import { useQuery } from '@tanstack/react-query';
+
+import { useRpc } from '_hooks';
+type GetCoinBalanceProps = {
+    address?: SuiAddress | null;
+    coinType: string;
+};
+export function useGetCoinBalance({ address, coinType }: GetCoinBalanceProps) {
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/hooks/useGetCoinBalance.ts:12]: Seems like you could probably get away with positional args here but not a deal breaker.

File: apps/wallet/src/ui/app/pages/home/tokens/TokensDetails.tsx [hunk 13]
```diff
@@ -76,13 +66,15 @@ function MyTokens({
 
 function TokenDetails({ coinType }: TokenDetailsProps) {
     const { loading, error, showError } = useObjectsState();
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/pages/home/tokens/TokensDetails.tsx:68]: Why do you still need this?

File: apps/wallet/src/ui/app/pages/home/tokens/TokensDetails.tsx [hunk 14]
```diff
@@ -30,38 +35,23 @@ const emptyWalletDescription = (
...
                             <CoinBalance
-                                type={aCoinType}
-                                balance={balances[aCoinType] || BigInt(0)}
-                                key={aCoinType}
+                                type={coinType}
+                                balance={totalBalance}
+                                key={coinType}
                             />
                         ))}
-                        {coinBalance <= 0 ? (
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/pages/home/tokens/TokensDetails.tsx:59]: Faucet button intentionally going away?

Comment (Jibz1) [reply apps/wallet/src/ui/app/pages/home/tokens/TokensDetails.tsx:59]: Thanks for catching it. this is fixed

File: apps/wallet/src/ui/app/pages/home/nft-transfer/TransferNFTForm.tsx [hunk 15]
```diff
@@ -40,7 +40,18 @@ function TransferNFTForm({
...
     }, [to]);
-    const maxGasCoinBalance = useIndividualCoinMaxBalance(SUI_TYPE_ARG);
+
+    const accountAddress = useAppSelector(({ account }) => account.address);
+    const { data: coinBalance, isLoading: loadingBalances } = useGetCoinBalance(
+        SUI_TYPE_ARG,
+        accountAddress
+    );
+
+    const maxGasCoinBalance = useMemo(
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/pages/home/nft-transfer/TransferNFTForm.tsx:50]: Doesn't need to be memo'd. We should only pull this out to optimize known expensive computation or to ensure referential stability for props.

File: apps/wallet/src/ui/app/staking/stake/StakingCard.tsx [hunk 16]
```diff
@@ -58,9 +56,16 @@ const initialValues = {
...
 
 function StakingCard() {
-    const coinType = GAS_TYPE_ARG;
+    const coinType = SUI_TYPE_ARG;
     const accountAddress = useAppSelector(({ account }) => account.address);
-    const aggregateBalances = useAppSelector(accountAggregateBalancesSelector);
+    const { data: suiBalance, isLoading: loadingSuiBalances } =
+        useGetCoinBalance(coinType, accountAddress);
+
+    const coinBalance = useMemo(() => {
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/staking/stake/StakingCard.tsx:64]: No need to memo: `const coinBalance = BigInt(suiBalance?.totalBalance || 0)`

File: apps/wallet/src/ui/app/pages/home/tokens/TokensDetails.tsx [hunk 17]
```diff
@@ -93,26 +88,29 @@ function TokenDetails({ coinType }: TokenDetailsProps) {
...
             {coinType && <PageTitle title={coinSymbol} back="/tokens" />}
 
-            <div className={st.container} data-testid="coin-page">
-                {showError && error ? (
-                    <Alert className={st.alert}>
+            <div
+                className="flex flex-col h-full flex-1 flex-grow items-center"
+                data-testid="coin-page"
+            >
+                {error instanceof Error ? (
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/pages/home/tokens/TokensDetails.tsx:95]: Why not just `!!error`? or check `isError`?

File: apps/wallet/src/ui/app/pages/home/tokens/TokenIconLink.tsx [hunk 18]
```diff
@@ -19,10 +19,13 @@ import { trackEvent } from '_src/shared/plausible';
...
 export function TokenIconLink({
     accountAddress,
+    noSuiToken,
 }: {
     accountAddress: SuiAddress;
+    noSuiToken?: boolean;
 }) {
-    const stakingEnabled = useFeature(FEATURES.STAKING_ENABLED).on;
+    const stakingEnabled =
+        useFeature(FEATURES.STAKING_ENABLED).on && !noSuiToken;
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/pages/home/tokens/TokenIconLink.tsx:28]: This should be removed, you could have no sui but have delegations you've created, and still want to view. Just revert this change

**Review by Jordan-Mysten — COMMENTED** [review]
few final issues, but should be good after those land

File: apps/wallet/src/ui/app/pages/home/tokens/TokensDetails.tsx (hunk 17 above)
Comment (Jibz1) [reply apps/wallet/src/ui/app/pages/home/tokens/TokensDetails.tsx:95]: Error is does not have a message without that. Unless the hook returns the type and Error, error.message is undefined.

File: apps/wallet/src/ui/app/pages/home/tokens/TokenIconLink.tsx (hunk 18 above)
Comment (Jibz1) [reply apps/wallet/src/ui/app/pages/home/tokens/TokenIconLink.tsx:28]: Ok.

File: apps/wallet/src/ui/app/pages/home/tokens/TokensDetails.tsx (hunk 17 above)
Comment (Jordan-Mysten) [reply apps/wallet/src/ui/app/pages/home/tokens/TokensDetails.tsx]: Yeah, it doesn't correctly, because the error could be any type (although this is changing in v5 of tanstack query).

File: apps/wallet/src/ui/app/pages/home/tokens/TokensDetails.tsx [hunk 19]
```diff
@@ -1,88 +1,80 @@
...
+    const { data: balance, isLoading: loadingBalances } =
+        useGetAllBalances(accountAddress);
 
-type TokensProps = {
-    allCoinTypes: string[];
-    coinBalance: bigint;
-    balances: Record<string, bigint>;
-    loading: boolean;
-};
+    const noSuiToken = useMemo(() => {
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/pages/home/tokens/TokensDetails.tsx:29]: Also doesn't need a memo

**Review by Jordan-Mysten — APPROVED** [review]
One nit but looks good

**Comment by mystie711** [comment]
Could we update the headers (SUI STAKE/MY COINS" color to "Sui Steel" and indent by 8px @Jibz1? Thanks!
